src/BeneficiariesFile.hs

```diff
@@ -106,3 +135,47 @@ readBeneficiariesFile :: Config -> IO [Beneficiary]
 readBeneficiariesFile conf = do
   raw <- readFile conf.beneficiariesFile
   pure $ either (error . Text.unpack) id (parseContent conf raw)
+
+prettyContent :: Config -> [Beneficiary] -> Either Text Text
+prettyContent conf =
+  first ("Beneficiaries file pretty printing error: " <>)
+    . fmap unlines
+    . mapM (prettyBeneficiary conf)
+
+prettyBeneficiary :: Config -> Beneficiary -> Either Text Text
+prettyBeneficiary conf (Beneficiary addr amt ac) =
+  unwords
+    <$> sequence
+      ( [prettyAddress conf addr]
+          <> [prettyAmount conf amt | isNothing conf.dropAmount]
+          <> [prettyAsset ac | isNothing conf.assetClass]
+      )
+
+prettyAmount :: Config -> Integer -> Either Text Text
+prettyAmount conf amt = Right $ maybe format (Text.pack . show) (scientificToMaybeInteger @Integer amt')
+  where
+    amt' = fromInteger @Scientific amt * (10 ^^ negate conf.decimalPlaces)
+
+    format = Text.pack . formatScientific Fixed Nothing $ amt'
+
+prettyAsset :: AssetClass -> Either Text Text
+prettyAsset ac
+  | ac == Value.assetClass Ada.adaSymbol Ada.adaToken = Right "lovelace"
+  | otherwise =
+    let (s, n) = Value.unAssetClass ac
+        sEncoded = encodeByteString . fromBuiltin . Value.unCurrencySymbol $ s
+        nEncoded = mapLeft (const "Token name was not UTF-8") . decodeUtf8' . fromBuiltin . Value.unTokenName $ n
+     in if n == ""
```

Comment:
I used `""` because n has type `TokenName`, and overloaded strings means so does `""`. Either way I've rearranged that code to get rid of the `mapLeft`, and patternmatch on `nEncoded` to get n as `Text` anyway, which cleans up the text concatting too

Reply:
Yeah that looks a lot cleaner, nice